src/routes/relay/entities/schema/sponsored-call.schema.helper.ts

```diff
@@ -7,39 +7,62 @@ import {
 } from '@safe-global/safe-deployments';
 import type { SingletonDeployment } from '@safe-global/safe-deployments';
 
+import { ERC20__factory } from '../../../../../contracts/openzeppelin';
+import {
+  Gnosis_safe__factory,
+  Multi_send__factory,
+  Proxy_factory__factory,
+} from '../../../../../contracts/safe/factories/v1.3.0';
+
 // ======================== General ========================
 
 /**
  * Checks the method selector of the call data to determine if it is the specified call
  * @param data call data
+ * @param fragment function fragment
  * @returns boolean
  */
-const isCalldata = (data: string, signature: string): boolean => {
-  const signatureId = ethers.id(signature).slice(0, 10);
-
-  return data.startsWith(signatureId);
+const isCalldata = (
+  data: string,
+  fragment: ethers.FunctionFragment,
+): boolean => {
+  return data.startsWith(fragment.selector);
 };
 
 // ======================== ERC-20 =========================
 
-const isErc20TransferCalldata = (data: string): boolean => {
-  const ERC20_TRANSFER_SIGNATURE = 'transfer(address,uint256)';
+const erc20Interface = ERC20__factory.createInterface();
+
+const ransferFragment = erc20Interface.getFunction('transfer');
 
-  return isCalldata(data, ERC20_TRANSFER_SIGNATURE);
+/**
+ * Checks the method selector of the call data to determine if it is a transfer call
+ * @param data call data
+ * @returns boolean
+ */
+const isErc20TransferCalldata = (data: string): boolean => {
+  return isCalldata(data, ransferFragment);
 };
 
+/**
+ * Extracts the `to` address from an ERC-20 transfer call
+ *
+ * @param data transfer call data
+ * @returns the `to` address of an ERC-20 transfer
+ */
 const getErc20TransferTo = (data: string): string => {
-  const ERC20_TRANFER_FRAGMENT = 'function transfer(address,uint256)';
-
-  const erc20Interface = new ethers.Interface([ERC20_TRANFER_FRAGMENT]);
-
-  const [to] = erc20Interface.decodeFunctionData(ERC20_TRANFER_FRAGMENT, data);
+  const [to] = erc20Interface.decodeFunctionData(ransferFragment, data);
 
   return to;
 };
 
 // ========================= Safe ==========================
 
+const safeInterface = Gnosis_safe__factory.createInterface();
+
+const setupFragment = safeInterface.getFunction('setup');
```

Comment:
nitpick: Given that this is only used in one place I'd move it there.

I guess this is mostly because this file has a lot of different functionality but if I think:

```typescript
const setupFunction = safeInterface.getFunction('setup');
const [owners] = safeInterface.decodeFunctionData(setupFunction, initializer);

```

provides more _inline_ context than: 

```typescript
const [owners] = safeInterface.decodeFunctionData(setupFragment, initializer);
```